autoload/test/javascript.vim

```diff
@@ -2,3 +2,13 @@ let test#javascript#patterns = {
   \ 'test': ['\v^\s*%(it|test)\s*[( ]\s*%("|'')(.*)%("|'')'],
   \ 'namespace': ['\v^\s*%(describe|suite|context)\s*[( ]\s*%("|'')(.*)%("|'')'],
 \}
+
+function! test#javascript#has_package(package) abort
+  for line in readfile("package.json")
```

Comment:
We need to be careful when searching for package.json recursively as we need to make the vendor directory ie `node_modules` is excluded.

I am inclined to leave this out of this PR as if we add recursive detection of package.json then we need to add recursive detection of test runners. Subdirectory support feels like a bigger feature, I know we parked it for PHP runners #49